Userland/DevTools/HackStudio/Git/GitWidget.cpp

```diff
@@ -71,45 +71,52 @@ GitWidget::GitWidget(const LexicalPath& repo_root)
         Gfx::Bitmap::try_load_from_file("/res/icons/16x16/minus.png").release_nonnull());
 }
 
-bool GitWidget::initialize()
+bool GitWidget::initialize(bool quiet)
 {
     auto result = GitRepo::try_to_create(m_repo_root);
     switch (result.type) {
     case GitRepo::CreateResult::Type::Success:
         m_git_repo = result.repo;
         return true;
     case GitRepo::CreateResult::Type::GitProgramNotFound:
-        GUI::MessageBox::show(window(), "Please install the Git port", "Error", GUI::MessageBox::Type::Error);
+        if (!quiet)
+            GUI::MessageBox::show(window(), "Please install the Git port", "Error", GUI::MessageBox::Type::Error);
         return false;
     case GitRepo::CreateResult::Type::NoGitRepo: {
-        auto decision = GUI::MessageBox::show(window(), "Create git repository?", "Git", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
-        if (decision != GUI::Dialog::ExecResult::ExecYes)
-            return false;
-        m_git_repo = GitRepo::initialize_repository(m_repo_root);
-        return true;
+        if (!quiet) {
+            auto decision = GUI::MessageBox::show(window(), "Create git repository?", "Git", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNo);
+            if (decision != GUI::Dialog::ExecResult::ExecYes)
+                return false;
+            m_git_repo = GitRepo::initialize_repository(m_repo_root);
+            return true;
+        }
+
+        return false;
```

Comment:
Please separate the addition of the `quiet` flag into its own commit, as it has nothing to do with the new widget.

Reply:
I feel like it is, since it is used directly with this new commit action: ``m_git_commit_action->set_enabled(m_git_widget->initialize_if_needed(true));``

Nonetheless, I will separate it now